library/cohesity_clone_vm.py

```diff
@@ -0,0 +1,392 @@
+        raise__cohesity_exception__handler(
+            "Failed to find the resource pool " + str(resource_pool), module)
+    except APIException as ex:
+        raise__cohesity_exception__handler(
+            str(json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def wait(cohesity_client, module):
+    '''
+    function to wait for clone task, waits for wait minutes passed to the module
+    :param cohesity_client: cohesity client
+    :param module: object that holds parameters passed to the module
+    :return:
+    '''
+    if module.params.get('wait_for_job'):
+        wait_time = module.params.get('wait_minutes')
+        while wait_time > 0:
+            clone_exist, clone_details = get_clone_task(
+                cohesity_client, module, True)
+            if not clone_exist:
+                return "The clone VMs request is accepted. Failed to check clone status during wait time"
+            elif clone_exist and clone_details.error:
+                raise__cohesity_exception__handler(
+                    "The clone VMs task failed", module)
+            elif clone_exist and clone_details.status == 'kFinished' and not clone_details.error:
+                return "The clone VMs task is successful"
+            time.sleep(SLEEP_TIME_SECONDS)
+            wait_time = wait_time - 1
+        return "The clone VMs request is accepted. The task is not finished in the wait time"
+    else:
+        return "The clone VMs request is accepted"
+
+
+def clone_vm(cohesity_client, module):
+    '''
+    function to clone the VMs
+    :param cohesity_client: cohesity client
+    :param module: object that holds parameters passed to the module
+    :return:
+    '''
+    try:
+        protection_job_details = get_protection_job_details(
+            cohesity_client, module)
+        objects = []
+        timestamp = module.params.get('backup_timestamp')
+        resource_pool = module.params.get('resource_pool')
+        for vm in module.params.get('vm_names'):
+            object_details = get_snapshot_details(
+                cohesity_client, module, timestamp, vm, protection_job_details.id)
+            objects.append(object_details)
+        clone_request = CloneTaskRequest()
+        clone_request.name = module.params.get('name')
+        clone_request.target_view_name = module.params.get('view_name')
+        clone_request.mtype = 'kCloneVMs'
+        clone_request.new_parent_id = protection_job_details.parent_source_id
+        clone_request.vmware_parameters = VmwareCloneParameters()
+        if module.params.get('suffix'):
+            clone_request.vmware_parameters.suffix = module.params.get('suffix')
+        if module.params.get('prefix'):
+            clone_request.vmware_parameters.prefix = module.params.get('prefix')
+        clone_request.vmware_parameters.powered_on = module.params.get(
+            'power_on')
```

Comment:
format

Reply:
autopep8 tool formatted it this way, may be I should check other tools.